op-chain-ops/cmd/withdrawals/main.go

```diff
@@ -33,60 +165,518 @@ func main() {
+							}
+							amount, ok := args[4].(*big.Int)
+							if !ok {
+								return fmt.Errorf("invalid abi")
+							}
+							extraData, ok := args[5].([]byte)
+							if !ok {
+								return fmt.Errorf("invalid abi")
+							}
+
+							log.Info(
+								"decoded calldata",
+								"l1Token", l1Token,
+								"l2Token", l2Token,
+								"from", from,
+								"to", to,
+								"amount", amount,
+								"extraData", extraData,
+							)
+
+							// Look for the ERC20 token transfer topic
+							for _, l := range receipt.Logs {
+								topic := l.Topics[0]
+								if topic == transferEvent.ID {
+									a, _ := transferEvent.Inputs.Unpack(l.Data)
+									// TODO: add a check here for balance diff
+									log.Info("EVENT FOUND", "args", a)
+								}
+								log.Info("receipt topic", "hex", topic.Hex())
+							}
+						case "finalizeETHWithdrawal":
+							// handle logic for ETH withdrawals
+							from, ok := args[0].(common.Address)
+							if !ok {
+								return fmt.Errorf("invalid type: from")
+							}
+							to, ok := args[1].(common.Address)
+							if !ok {
+								return fmt.Errorf("invalid type: to")
+							}
+							amount, ok := args[2].(*big.Int)
+							if !ok {
+								return fmt.Errorf("invalid type: amount")
+							}
+							extraData, ok := args[3].([]byte)
+							if !ok {
+								return fmt.Errorf("invalid type: extraData")
+							}
+
+							log.Info(
+								"decoded calldata",
+								"from", from,
+								"to", to,
+								"amount", amount,
+								"extraData", extraData,
+							)
+
+						default:
+							log.Info("Unhandled method", "name", method.Name)
+						}
+					}
+
+					// Ensure that the target's balance was increasedData correctly
+					wdValue, err := wd.Value()
+					if err != nil {
+						return err
+					}
+					if method != nil {
+						log.Info("withdrawal action", "function", method.Name, "value", wdValue)
+					} else {
+						log.Info("unknown method", "to", wd.Target, "data", hexutil.Encode(wd.Data))
+
+						badWithdrawals = append(badWithdrawals, badWithdrawal{
+							Withdrawal: withdrawal,
+							Legacy:     wd,
+							Trace:      finalizationTrace,
+							Index:      i,
+						})
+
+					}
+					// check that the user's intents are actually executed
+					if common.HexToAddress(callFrame.To) != *wd.Target {
+						badWithdrawals = append(badWithdrawals, badWithdrawal{
+							Withdrawal: withdrawal,
+							Legacy:     wd,
+							Trace:      finalizationTrace,
+							Index:      i,
+						})
+
+						log.Info("target mismatch", "index", i)
+
+						continue
+					}
+					if !bytes.Equal(hexutil.MustDecode(callFrame.Input), wd.Data) {
+						badWithdrawals = append(badWithdrawals, badWithdrawal{
+							Withdrawal: withdrawal,
+							Legacy:     wd,
+							Trace:      finalizationTrace,
+							Index:      i,
+						})
+
+						log.Info("calldata mismatch", "index", i)
+
+						continue
+					}
+					if callFrame.Value != "0x"+wdValue.Text(16) {
```

Comment:
do `callFrame.Value` and `wdValue.Text(16)` pad with any zeroes, or are they just hex encoded ints? Need to be sure unexpected padding cannot throw off the comparison.